crates/config/src/app.rs

```diff
@@ -0,0 +1,235 @@
+use crate::agent::AgentConfig;
+use crate::keys::Libp2pKeyWrapper;
+use crate::network::NetworkConfig;
+use crate::signing::{CommitteeConfig, UnvalidatedCommitteeConfig};
+use figment::Figment;
+use figment::providers::{Format, Json, Toml};
+use libp2p::Multiaddr;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use shellexpand::tilde;
+use std::path::{Path, PathBuf};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct AppConfig {
+    pub agent: AgentConfig,
+    pub networks: Vec<NetworkConfig>,
+    pub libp2p: Libp2pConfig,
+    pub committee: CommitteeConfig,
+}
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Libp2pConfig {
+    pub secret_key: Libp2pKeyWrapper,
+    pub multiaddr: Multiaddr,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct ConfigFile {
+    pub agent: AgentConfig,
+    pub networks: Vec<NetworkConfig>,
+    pub libp2p: Libp2pConfig,
+    pub committee: UnvalidatedCommitteeConfig,
+}
+
+pub fn load_app_config(config_path: String) -> anyhow::Result<AppConfig> {
+    println!("Loading app config from {}", config_path);
+
+    let path = Path::new(&config_path);
+    let config_file = match path.extension().and_then(|s| s.to_str()) {
+        Some("toml") => Figment::new().merge(Toml::file(path)),
+        Some("json") => Figment::new().merge(Json::file(path)),
+        _ => anyhow::bail!("unsupported config format"),
+    };
```

Comment:
I'm not sure what the best approach is here tbh, should we filter by extension, or go with the `CommitteeConfig`-style approach using figment + toml + json merges:
```rust
Figment::new()
            .merge(Toml::file(&path))
            .merge(Json::file(&path))
            .extract()?;
```

Reply:
Hmm I think this might be an unused carryover 🤔 I vaguely remember using this, but it would blow up because it would try and parse `path` as both toml _and_ json